elements/camera-capture/settings-pane.js

```diff
@@ -0,0 +1,291 @@
+    controls.forEach(control => {
+      control.onchange = e => {
+        const id = e.target.getAttribute('id');
+        const value = e.detail.value;
+        console.log(id, ":", value);
+
+        let constraints = { advanced: [{}] };
+        constraints.advanced[0][id] = value;
+        if (id == 'exposureTime') {
+          constraints.advanced[0]['exposureMode'] = 'manual';
+        } else if (id == 'focusDistance') {
+          constraints.advanced[0]['focusMode'] = 'manual';
+        } else if (id == 'colorTemperature') {
+          this._colorTemperatureChange(value);
+          constraints.advanced[0]['whiteBalanceMode'] = 'manual';
+        }
+
+        this.dispatchEvent(new CustomEvent('constraintschange', { detail: { constraints } }));
+      };
+    });
+  }
+
+  reset() {
+    const id = this.activePane.getAttribute("id");
+    let constraints = { advanced: [{}] };
+
+    if (id == 'standardSettings') {
+      Array.from(this.activePane.children).forEach(child => {
+        child.value = (child.max - child.min) / 2 + Number(child.min);
+        if (!child.disabled) {
+          constraints.advanced[0][id] = child.value;
+        }
+      });
+    } else if (id == 'exposureTimeSettings') {
+      constraints.advanced[0]['exposureMode'] = 'continuous';
+      this.activePane.children[0].value = 0;
+    } else if (id == 'focusDistanceSettings') {
+      constraints.advanced[0]['focusMode'] = 'continuous';
+      this.activePane.children[0].value = 0;
+    } else if (id == 'colorTemperatureSettings') {
+      constraints.advanced[0]['whiteBalanceMode'] = 'continuous';
+      this.activePane.children[0].value = 0;
+      this._colorTemperatureChange(0);
+    } else if (id == 'zoomSettings') {
+      constraints.advanced[0]['zoom'] = 1;
+      this.activePane.children[0].value = 1;
+    }
+
+    // The handler of 'constrainstchange' should call applyFromTrack after applying
+    // reset settings.
+    this.dispatchEvent(new CustomEvent('constraintschange', { detail: { constraints } }));
+  }
+
+  render() {
+    return html`
+      <link href="../css/google-icons.css" rel="stylesheet">
+      <div id="settings-bar" @click=${e => e.stopPropagation()}>
+        <div id="isoSettings" class="pro-settings hidden">
+          <settings-slider label="ISO" id="iso" min="0" max="10" value="0" step="1"></settings-slider>
+        </div>
+        <div id="exposureTimeSettings" class="pro-settings hidden">
+          <settings-slider metering label="Exposure" id="exposureTime" min="0" max="10" value="0" step="1"></settings-slider>
+        </div>
+        <div id="focusDistanceSettings" class="pro-settings hidden">
+          <settings-slider metering label="Focus distance" id="focusDistance" min="0" max="10" value="0" step="1"></settings-slider>
+        </div>
+        <div id="standardSettings" class="pro-settings hidden">
+          <settings-slider label="Contrast" id="contrast" min="0" max="10" value="0" step="1"></settings-slider>
```

Comment:
remove defaults? we are reading it in [L.108](https://github.com/riju/WebCamera/pull/35/files#diff-11d3a1dff56a6f75cf2787ec072c405eR108)

Reply:
Done